src/EnergyPlus/DaylightingDevices.cc

```diff
@@ -1620,6 +1614,164 @@ namespace DaylightingDevices {
+        EnergyPlusData &state, Real64 &viewFactorToShelf, Real64 &viewFactorToSky, Real64 &viewFactorToGround, int WinSurf, int ShelfNum)
+    {
+        // First, make sure none of the view factors are less than zero and return if there isn't a problem or if
+        // view factor to shelf greater than one.  Both cases together would also eliminate if other views are zero
+        // which means nothing would need to be done.
+        if (viewFactorToSky <= 0.0) viewFactorToSky = 0.0;
+        if (viewFactorToGround <= 0.0) viewFactorToGround = 0.0;
+        if (viewFactorToShelf <= 0.0) { // No shelf impact for which to account
+            ShowWarningError(state,
+                             format("DaylightingDevice:Shelf = {}:  Window view factor to shelf was less than 0.  This should not happen.",
+                                    state.dataDaylightingDevicesData->Shelf(ShelfNum).Name));
+            ShowContinueError(state, "The view factor has been reset to zero.");
+            viewFactorToShelf = 0.0;
+            if ((viewFactorToGround + viewFactorToSky) > 1.0) { // This data came in incorrect, fix by proportional reduction
+                viewFactorToGround = viewFactorToGround / (viewFactorToGround + viewFactorToSky);
+                viewFactorToSky = 1.0 - viewFactorToGround;
+                ShowWarningError(state,
+                                 format("DaylightingDevice:Shelf = {}:  The sum of the window view factors to ground and sky were greater than 1.  "
+                                        "This should not happen.",
+                                        state.dataDaylightingDevicesData->Shelf(ShelfNum).Name));
+                ShowContinueError(
+                    state, "The view factors have been reset to so that they do not exceed 1.  Check/fix your input file data to avoid this issue.");
+            }
+            return;
+        }
+        if (viewFactorToShelf + viewFactorToSky + viewFactorToGround <= 1.0) return; // nothing wrong here
+        if (viewFactorToShelf >= 1.0) { // Don't allow shelf view of greater than 1 (zero out other views)
+            ShowWarningError(state,
+                             format("DaylightingDevice:Shelf = {}:  Window view factor to shelf was greater than 1.  This should not happen.",
+                                    state.dataDaylightingDevicesData->Shelf(ShelfNum).Name));
+            ShowContinueError(state, "The view factor has been reset to 1 and the other view factors to sky and ground have been set to 0.");
+            viewFactorToShelf = 1.0;
+            viewFactorToGround = 0.0;
+            viewFactorToSky = 0.0;
+            return;
+        }
+
+        // If the flow is still here, there is something that needs to be adjusted so set the maximum shelf height and the minimum window height
+        int ShelfSurf = state.dataDaylightingDevicesData->Shelf(ShelfNum).OutSurf;
+        Real64 zShelfMax = -9999.9;
+        Real64 zShelfMin = 9999.9;
```

Comment:
While this would be pretty rare given the values set here, I agree that it makes more sense to use the first vertex to set the Max and Min values and then go from there.  This will be changed in the next commit.

Reply:
@RKStrand Sorry here it should be `state.dataDaylightingDevicesData->Shelf(ShelfNum).Vertex(1).z` as the vertex index start with 1 instead of zero.

Reply:
No problem.  I knew what you meant.  state.dataSurface->Surface(ShelfSurf).Vertex(1).z works in this case.  I went with that.

Reply:
Also, then the loop can start with `for(int vertex = 2;...`---the first one no longer needs to be compared.

Reply:
Yes.  I caught that.  All of the changes have now been committed.  Hoping for a clean CI and that all of the concerns have been addressed now.  Thanks for your review!